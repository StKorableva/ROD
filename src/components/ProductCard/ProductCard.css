.productCard{
    position:relative;
    background-color: #000000;
    margin:18px;
    height: 322px;
    width: 158px;
}
.productCard__img{
    height: 254px;
    width: 158px;
}

.productCard p{
    text-align:left;
    color:white;
    font-size: 14px;
}
.productCard__addToFavorites{
    margin:5px;
    position:absolute;
    top:0;
    right: 0;
}