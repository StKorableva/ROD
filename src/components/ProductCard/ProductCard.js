import './ProductCard.css';
import heartIcon from '../../img/heart-icon.svg';

function ProductCard(props) {
	let addToFavoritesClickHandler = function (e) {
		e.stopPropagation();
		props.addToFavorites();
	};
	return (
		<div className="productCard" onClick={props.openProductPage}>
			<div className="productCard__addToFavorites" onClick={addToFavoritesClickHandler}>
				<img className="productCard__addToFavorites__icon" src={heartIcon} alt=""></img>
			</div>
			<img className="productCard__img" src={props.url} alt="houston we have a problems"></img>
			<p className="productCard__text">{props.text}</p>
			<p className="productCard__price">{props.price}</p>
		</div>
	);
}

export default ProductCard;
