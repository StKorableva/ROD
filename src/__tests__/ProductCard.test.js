import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import ProductCard from "../components/ProductCard/ProductCard";

Enzyme.configure({ adapter: new Adapter() });
describe("base test", () => {
	let component = shallow(
		<ProductCard
			openProductPage={() => {
				console.log("callback#1");
			}}
			addToFavorites={() => {
				console.log("callback#2");
			}}
			url="./test"
			text="test data / style for test(without fonts)"
			price="1000"
		></ProductCard>
	);

	it("render test", () => {
		expect(component).toHaveLength(1);
	});

	it("snapshot test", () => {
		expect(toJSON(component)).toMatchSnapshot();
	});
});
