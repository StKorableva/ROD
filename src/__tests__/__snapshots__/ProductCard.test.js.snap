// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base test snapshot test 1`] = `
<div
  className="productCard"
  onClick={[Function]}
>
  <div
    className="productCard__addToFavorites"
    onClick={[Function]}
  />
  <img
    alt="houston we have a problems"
    className="productCard__img"
    src="./test"
  />
  <p
    className="productCard__text"
  >
    test data / style for test(without fonts)
  </p>
  <p
    className="productCard__price"
  >
    1000
    â‚½
  </p>
</div>
`;
